x-pack/filebeat/input/httpjson/config_oauth.go

```diff
@@ -0,0 +1,161 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package httpjson
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+	"golang.org/x/oauth2/endpoints"
+	"golang.org/x/oauth2/google"
+)
+
+type OAuth2Provider string
+
+const (
+	OAuth2ProviderDefault OAuth2Provider = ""
+	OAuth2ProviderAzure   OAuth2Provider = "azure"
+	OAuth2ProviderGoogle  OAuth2Provider = "google"
+)
+
+func (p OAuth2Provider) canonical() OAuth2Provider {
+	return OAuth2Provider(strings.ToLower(string(p)))
+}
+
+// OAuth2 contains information about oauth2 authentication settings.
+type OAuth2 struct {
+	// common oauth fields
+	ClientID       string              `config:"client_id"`
+	ClientSecret   string              `config:"client_secret"`
+	Enabled        *bool               `config:"enabled"`
+	EndpointParams map[string][]string `config:"endpoint_params"`
+	Provider       OAuth2Provider      `config:"provider"`
+	Scopes         []string            `config:"scopes"`
+	TokenURL       string              `config:"token_url"`
+
+	// google specific
+	GoogleCredentialsFile string `config:"google_credentials_file"`
+	GoogleCredentialsJSON []byte `config:"google_credentials_json"`
+	GoogleJWTFile         string `config:"google_jwt_file"`
+
+	// microsoft azure specific
+	AzureTenantID string `config:"azure_tenant_id"`
```

Comment:
```suggestion
	AzureTenantID string `config:"azure.tenant_id"`
```

(to be consistent with `google.X`)